### PR TITLE
Containers refactor III - This time its composable

### PR DESCRIPTION
**Pull request by pete-watters**
> Try out Leather build d804675 — [Extension build](https://github.com/leather-io/extension/actions/runs/10200659814), [Test report](https://leather-io.github.io/playwright-reports/chore-refactor-containers), [Storybook](https://chore-refactor-containers--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=chore-refactor-containers)

This PR addresses some concerns around how I implemented adding global containers. [See discussion here](https://github.com/orgs/leather-io/discussions/87). 

- The original PR I added was https://github.com/leather-io/extension/pull/4655
- After another attempt using `Context` [here](https://github.com/leather-io/extension/pull/5619), now each page is responsible for defining it's own header via composition.

There is a lot of code here but the main update is:
- use of routes to configure `app/features/container` is now gone 
- Pages get have to opt in and configure their own header  - `<MainHeader`, `<PageHeader`, `<PopupHeader`
- `<Home` pages have this set @ `app-route` level as they share a simpler header 
- Other full view pages use the `<Page` wrapper to control their width:
```
<Container>
    <PageHeader title="Send" />
    <ContainerLayout>
          <Content>
                <Page>
                </Page>
          </Content>
      </ContainerLayout>    
   </Container>     
```
- `<Popup` pages are popup windows for Approval UX and don't need the `<Page ` wrapper 

This is an improvement but could probably be improved more so please let me know of any suggestions.

### CODE REVIEW

File: src/app/components/broadcast-error-dialog/broadcast-error-dialog.tsx [hunk 1]
```diff
@@ -6,8 +6,8 @@ import get from 'lodash.get';
 
 import { Button, Dialog } from '@leather.io/ui';
 
-import { Footer } from '@app/ui/components/containers/footers/footer';
-import { DialogHeader } from '@app/ui/components/containers/headers/dialog-header';
+import { Footer } from '@app/ui/layout/containers/footers/footer';
+import { DialogHeader } from '@app/ui/layout/containers/headers/dialog-header';
```
Comment (pete-watters) [inline src/app/components/broadcast-error-dialog/broadcast-error-dialog.tsx:10]: I need to move `DialogHeader` to the monorepo alongside `Dialog`

File: src/app/components/broadcast-error-dialog/broadcast-error-dialog.tsx [hunk 2]
```diff
@@ -6,8 +6,8 @@ import get from 'lodash.get';
 
 import { Button, Dialog } from '@leather.io/ui';
 
-import { Footer } from '@app/ui/components/containers/footers/footer';
-import { DialogHeader } from '@app/ui/components/containers/headers/dialog-header';
+import { Footer } from '@app/ui/layout/containers/footers/footer';
```
Comment (pete-watters) [inline src/app/components/broadcast-error-dialog/broadcast-error-dialog.tsx:9]: I need to merge `'@app/ui/layout/containers` with `'@app/containers/layouts`

File: src/app/features/container/headers/header.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  hideLogo?: boolean;
+}
+
+export function Header({ hideBackButton = false, hideLogo = false }: HeaderProps) {
+  const { chain, name: chainName } = useCurrentNetworkState();
+  const [isShowingSwitchAccount, setIsShowingSwitchAccount] = useState(false);
+
+  const navigate = useNavigate();
+  return (
+    <>
```
Comment (pete-watters) [inline src/app/features/container/headers/header.tsx:30]: I need to do some refactors of `Header` files to make the code more DRY and share the layout

**Comment by markmhendrickson** [comment]
Thanks for taking this initiative, @pete-watters. Is it worth getting some reviewer eyes on this now for feedback, despite it being a draft PR still?

**Comment by pete-watters** [comment]
> Thanks for taking this initiative, @pete-watters. Is it worth getting some reviewer eyes on this now for feedback, despite it being a draft PR still?

I'm working on a final fix for Popup's then I will open for full review in the next hour no matter what 👍

**Comment by pete-watters** [comment]
> > Thanks for taking this initiative, @pete-watters. Is it worth getting some reviewer eyes on this now for feedback, despite it being a draft PR still?
> 
> I'm working on a final fix for Popup's then I will open for full review in the next hour no matter what 👍

OK it turns out everything was working and the problem was solved by a lunch break

**Comment by pete-watters** [comment]
@kyranjamie @alter-eggo : Can you please check this PR when you have some time? 

Its quite large so apologies for that but hopefully it adds a big improvement to the scalability of the container pattern

File: src/app/features/container/headers/main.header.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { HeaderActionButton } from '@app/components/layout/headers/header-action-button';
+import { HeaderGrid, HeaderGridRightCol } from '@app/components/layout/headers/header-grid';
+import { LogoBox } from '@app/components/layout/headers/logo-box';
+import { Settings } from '@app/features/settings/settings';
+
+interface MainHeaderProps {
+  hideBackButton?: boolean;
+  hideLogo?: boolean;
+}
+
```
Comment (pete-watters) [inline src/app/features/container/headers/main.header.tsx:20]: I can refactor `<MainHeader` and `<PageHeader` to merge them and keep them DRY but didn't want to let this drag on longer before review

File: src/app/pages/onboarding/sign-in/sign-in.tsx [hunk 5]
```diff
@@ -18,29 +19,34 @@ export function SignIn() {
...
+    <>
+      <MainHeader />
+      <Content>
+        <TwoColumnLayout
+          title={
+            <>
+              Sign in <br /> with your <br /> Secret Key
+            </>
+          }
+          content={<>Speed things up by pasting your entire Secret Key in one go.</>}
```
Comment (kyranjamie) [inline src/app/pages/onboarding/sign-in/sign-in.tsx]: Can just be a string without fragment

**Review by kyranjamie — APPROVED** [review]
Thanks Pete, this looks like a well organised, configurable solution 👍🏻

File: src/app/pages/onboarding/sign-in/sign-in.tsx (hunk 5 above)
Comment (pete-watters) [reply src/app/pages/onboarding/sign-in/sign-in.tsx]: Good spot. I've pushed a commit to update that [here](https://github.com/leather-io/extension/pull/5715/commits/d8046751d6a43270caa5775f1b55b9b16ce41b01)

**Pull request merged**